go/tools/gazelle/rules/construct.go

```diff
@@ -82,30 +87,73 @@ func newValue(val interface{}) (bzl.Expr, error) {
 			list = append(list, elem)
 		}
 		return &bzl.ListExpr{List: list}, nil
-	case reflect.Struct:
-		glob, ok := val.(globvalue)
-		if !ok {
-			return nil, fmt.Errorf("not implemented %T", val)
+
+	case reflect.Map:
+		rkeys := rv.MapKeys()
+		sort.Slice(rkeys, func(i, j int) bool {
+			return rkeys[i].String() < rkeys[j].String()
+		})
+		args := make([]bzl.Expr, 0, len(rkeys))
```

Comment:
The only path out of here that does not set every element of args abandons it totally, so you might as well just make it the right size and avoid append.
	args := make([]bzl.Expr, len(rkeys))
	for i, rk := range rkeys {
		k := &bzl.StringExpr{Value: rk.String()}
		v, err := newValue(rv.MapIndex(rk))
		if err != nil {
			return nil, err
		}
		args[i] = &bzl.KeyValueExpr{Key: k, Value: v})
	}